setup.py

```diff
@@ -7,9 +7,21 @@
 README = open(os.path.join(here, 'README.md')).read()
 CHANGES = open(os.path.join(here, 'CHANGES.txt')).read()
 
-REQUIREMENTS_FILE = 'requirements-py3.txt'
-if sys.version_info < (3, 0, 0):
-    REQUIREMENTS_FILE = 'requirements.txt'
+REQUIREMENTS_FILE = 'requirements.txt'
+
+# As pipenv install -e edits the pipfile itself, it is better to avoid chances of it being called
+# further in the code. Thus we create a simple requirements file to avoid unintended consequences
+
+if not os.path.exists(os.path.join(here, REQUIREMENTS_FILE)):
+    requirements = open(os.path.join(here, 'Pipfile')).read().split('\n')
+    requirements = requirements[requirements.index('[packages]') + 1:
+                                requirements.index('[requires]')]
+    requirements.pop(requirements.index('[dev-packages]'))
+    requirements = [req.replace('"', '') for req in requirements]
+    requirements = [req.replace(' = ', '') for req in requirements]
+    with open(os.path.join(here, REQUIREMENTS_FILE), 'w+') as file:
+        for req in requirements:
+            file.write(f"{req}\n")
```

Comment:
This could be much simpler, simply call `pipenv lock -r` no need to parse pipfile !

```python
import subprocess

requires = subprocess.check_output('pipenv lock -d -r', shell=True, stderr=None).decode().split('\n')
```

Then no need to generate a local file and add it to git ignore and dockerignore. Moreover the code above would be problematic when the file already exists it won't be updated !

Reply:
I didn't know about the -r flag, thanks for the input, it will indeed greatly simplify this.